trait - simplify method with null coalescing operator

k-timoshenko opened the pull request.

k-timoshenko committed:
trait - simplify method with null coalescing operator

MatthewPattell reviewed (commented):
Add please return type for all methods :)

k-timoshenko committed:
trait - refactor - declare return types where possible

MatthewPattell commented on ImplementServices.php:
```diff
@@ -77,31 +77,30 @@ public function __call($name, $params)
-    private function getService(string $name, array $params = [])
+    private function getService(string $name, array $params = []): ?BaseService
     {
         $className = $this->getClassName($name);
```
add a line break

MatthewPattell commented on ImplementServices.php:
```diff
@@ -77,31 +77,30 @@ public function __call($name, $params)
-
-            return $this->servicesInstances[$name];
+        if (!isset($this->servicesInstances[$name])) {
+            $this->servicesInstances[$name] = new $className($this, isset($params[0]) && is_array($params[0]) ? $params[0] : []);
```
Can shorten this line: is_array($params[0] ?? '')
Thx

MatthewPattell commented on ImplementServices.php:
```diff
@@ -77,31 +77,30 @@ public function __call($name, $params)
-    private function getService(string $name, array $params = [])
+    private function getService(string $name, array $params = []): ?BaseService
     {
         $className = $this->getClassName($name);
```
add a line break ^)

MatthewPattell replied:
And change php version in composer: 7.1 (Nullable RFC)

k-timoshenko committed:
trait - reformat - fix code style

k-timoshenko committed:
composer - set correct version for php and yii

- php 7.1 required for nullable return type hit
- yii introduced BaseObject in 2.0.13 version

k-timoshenko commented:
Done.
php now min 7.1
yii 2.0.12 because of BaseObject
